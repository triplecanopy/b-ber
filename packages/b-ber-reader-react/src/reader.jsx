import React from 'react'
import ConnectedApp from '.'

render(
  <ConnectedApp props={window.__SERVER_DATA__ || {}} />,
  document.getElementById('root')
)
