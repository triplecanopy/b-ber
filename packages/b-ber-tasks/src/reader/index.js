/* eslint-disable camelcase */
/* eslint-disable class-methods-use-this */

import fs from 'fs-extra'
import path from 'path'
import crypto from 'crypto'
import state from '@canopycanopycanopy/b-ber-lib/State'
import find from 'lodash/find'
import { Url } from '@canopycanopycanopy/b-ber-lib'
import { generateWebpubManifest } from '@canopycanopycanopy/b-ber-lib/utils'
import log from '@canopycanopycanopy/b-ber-logger'
import rrdir from 'recursive-readdir'
import has from 'lodash/has'

class Reader {
  constructor() {
    this.outputDirName = 'epub'
    this.outputDir = state.dist.root(this.outputDirName)
    this.apiDir = state.dist.root('api')
    this.epubAssets = ['META-INF', 'OPS', 'mimetype']

    this.readerModuleName = '@canopycanopycanopy/b-ber-reader'
    this.readerModuleDistDir = 'dist'
    this.readerAppPath = null

    // eslint-disable-next-line no-constructor-return
    return (
      this.createOutputDirs()
        .then(() => this.ensureReaderModuleExists())
        .then(() => this.copyEpubToOutputDir())
        .then(() => this.writeBookManifest())
        .then(() => this.writeWebpubManifest())
        .then(() => this.copyReaderAppToOutputDir())
        .then(() => this.injectServerDataIntoTemplate())
        // .then(() => this.updateLinkedResourcesWithAbsolutePaths())
        .then(() => this.updateAssetURLsWithAbsolutePaths())
        .then(() => this.injectWebpubManifestLink())
        .catch(log.error)
    )
  }

  get remoteURL() {
    if (
      process.env.NODE_ENV === 'production' &&
      (!state.config ||
        !state.config.remote_url ||
        /^http/.test(state.config.remote_url) === false)
    ) {
      throw new Error(
        'Task [build/reader] requires a remote_url to be set in config.yml'
      )
    }
    return state.config.remote_url || 'http://localhost:4000/'
  }

  createDirname(s) {
    if (!s || typeof s !== 'string') {
      return crypto.randomBytes(20).toString('hex')
    }
    return s.replace(/[^0-9a-zA-Z-]/g, '-')
  }

  ensureReaderModuleExists() {
    try {
      this.readerAppPath = path.join(
        path.dirname(path.join(require.resolve(this.readerModuleName))),
        this.readerModuleDistDir
      )
      return
    } catch (err) {
      // module not found using require.resolve, so we check if there's a symlinked version available
      log.warn(
        `Could not find globally installed module ${this.readerModuleName}`
      )
    }

    const { paths } = module
    let modulePath
    for (let i = 0; i < paths.length; i++) {
      const _path = path.resolve(paths[i], this.readerModuleName)
      if (fs.existsSync(_path)) {
        modulePath = _path
        break
      }
    }

    if (!modulePath) {
      log.error(
        `Cannot find module ${this.readerModuleName}. Try running npm i -S ${this.readerModuleName}`
      )
    }

    try {
      this.readerAppPath = fs.realpathSync(
        path.join(modulePath, this.readerModuleDistDir)
      )
      const pkg = fs.readJsonSync(
        path.join(modulePath, this.readerModuleDistDir, 'package.json')
      )
      log.warn(`Loaded ${this.readerModuleName} v${pkg.version}`)

      return
    } catch (err) {
      log.error(`
                A symlinked version of ${this.readerModuleName} was found but is inaccessible.
                Try running npm i -S ${this.readerModuleName}, or rebuilding the reader package if running this command in a development environment
            `)
      process.exit(1)
    }
  }

  createOutputDirs() {
    return fs.ensureDir(this.outputDir).then(() => fs.ensureDir(this.apiDir))
  }

  copyEpubToOutputDir() {
    const epubDir = this.createDirname(this.getBookMetadata('identifier'))
    const promises = this.epubAssets.map(item =>
      fs.move(state.dist.root(item), path.join(this.outputDir, epubDir, item))
    )

    return Promise.all(promises)
  }

  getBookMetadata(term) {
    if (!term) return state.metadata.json()

    const entry = find(state.metadata.json(), { term })
    if (entry && entry.value) return entry.value

    log.warn(`Could not find metadata value for ${term}`)

    return ''
  }

  getProjectConfig(term) {
    if (!term) return state.config

    if (!has(state.config, term)) {
      log.warn(`Invalid property for config: ${term}`)
    }

    return state.config[term]
  }

  writeBookManifest() {
    // Get metadata for books.json
    const id = this.getBookMetadata('identifier')
    const title = this.getBookMetadata('title')

    const url = `${Url.trimSlashes(this.remoteURL)}/${
      this.outputDirName
    }/${this.createDirname(id)}`

    const cover = `${url}/OPS/images/${this.getBookMetadata('cover')}`

    // Get config required by reader for books.json
    const { downloads, ui_options, layout } = this.getProjectConfig()

    const manifest = [{ title, url, cover, id, downloads, ui_options, layout }]

    // Write to an `api` dir in case the app is being deployed statically
    return fs.writeJson(path.join(this.apiDir, 'books.json'), manifest)
  }

  writeWebpubManifest() {
    const assetsDir = path.join(
      process.cwd(),
      this.outputDir,
      this.getBookMetadata('identifier'),
      'OPS'
    )
    return rrdir(assetsDir).then(files => {
      const manifest = generateWebpubManifest(files)
      fs.writeJson(state.dist.root('manifest.json'), manifest)
    })
  }

  injectWebpubManifestLink() {
    const indexHTML = state.dist.root('index.html')
    const readerURL = Url.addTrailingSlash(this.getProjectConfig('reader_url'))

    let contents
    contents = fs.readFileSync(indexHTML, 'utf8')
    contents = contents.replace(
      /<\/head>/,
      `<link rel="manifest" type="application/webpub+json" href="${readerURL}manifest.json"></head>`
    )

    return fs.writeFile(indexHTML, contents)
  }

  copyReaderAppToOutputDir() {
    const promises = fs
      .readdirSync(this.readerAppPath)
      .map(file =>
        fs.copy(
          path.join(this.readerAppPath, file),
          path.resolve(state.dist.root(file))
        )
      )

    return Promise.all(promises)
  }

  injectServerDataIntoTemplate() {
    const indexHTML = state.dist.root('index.html')
    const readerURL = Url.addTrailingSlash(this.getProjectConfig('reader_url'))
    const identifier = this.getBookMetadata('identifier')
    const bookURL = `${readerURL}epub/${identifier}`
    const serverData = {
      books: [
        {
          title: this.getBookMetadata('title'),
          url: bookURL,
          cover: this.getBookMetadata('cover'),
        },
      ],
      bookURL,
      projectURL: Url.addTrailingSlash(this.getProjectConfig('remote_url')),
      downloads: this.getProjectConfig('downloads'),
      basePath: Url.addTrailingSlash(this.getProjectConfig('base_path')),
      loadRemoteLibrary: false,
      uiOptions: this.getProjectConfig('ui_options'),
      cache: this.getProjectConfig('cache'),
      layout: this.getProjectConfig('layout'),
    }

    let contents
    contents = fs.readFileSync(indexHTML, 'utf8')
    contents = contents.replace(
      /__SERVER_DATA__ = {}/,
      `__SERVER_DATA__ = ${JSON.stringify(serverData)}`
    )

    return fs.writeFile(indexHTML, contents)
  }

  // Update URLs in CSS
  updateLinkedResourcesWithAbsolutePaths() {
    const indexContents = fs.readFileSync(state.dist.root('index.html'), 'utf8')
    const versionHash = indexContents.match(/link href="\/(\w+\.css)"/)[1]
    const stylesheet = state.dist.root(versionHash)
    const readerURL = Url.addTrailingSlash(this.getProjectConfig('reader_url'))

    let contents
    contents = fs.readFileSync(stylesheet, 'utf8')
    contents = contents.replace(/url\(\//g, `url(${readerURL}`)

    return fs.writeFile(stylesheet, contents)
  }

  updateAssetURLsWithAbsolutePaths() {
    const indexHTML = state.dist.root('index.html')
    const readerURL = Url.removeTrailingSlash(
      this.getProjectConfig('reader_url')
    )

    let contents
    contents = fs.readFileSync(indexHTML, 'utf8')
    contents = contents.replace(
      /(src|href)="(\/[^"]+?)"/g,
      `$1="${readerURL}$2"`
    )

    return fs.writeFile(indexHTML, contents)
  }
}

const main = () => new Reader()

export default main
