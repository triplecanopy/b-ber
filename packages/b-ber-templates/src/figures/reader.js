import { media, iframe, figure, vimeo } from './helpers'

const reader = {
  landscape: data =>
    figure({ data, ratioName: 'landscape', applyInlineClasses: true }),
  portrait: data =>
    figure({ data, ratioName: 'portrait', applyInlineClasses: true }),
  'portrait-high': data =>
    figure({ data, ratioName: 'portrait-high', applyInlineClasses: true }),
  square: data =>
    figure({ data, ratioName: 'square', applyInlineClasses: true }),
  audio: data => media({ ...data, applyInlineClasses: true }),
  video: data => media({ ...data, applyInlineClasses: true }),
  iframe: data => iframe({ ...data, applyInlineClasses: true }),
  vimeo: data => vimeo({ ...data, applyInlineClasses: true }),
}

export default reader
