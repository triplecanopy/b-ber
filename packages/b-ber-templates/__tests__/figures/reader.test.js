import fs from 'fs-extra'
import reader from '../../src/figures/reader'

jest.mock('@canopycanopycanopy/b-ber-lib/State', () => ({}))

afterAll(() => fs.remove('_project'))

describe('templates.reader', () => {
  test('creates the markup', () => {
    const data = {
      inline: true,
      classes: 'test',
      id: 1,
      ref: 1,
      alt: 'test',
      source: 'test',
      caption: 'test',
      attrString: 'autoplay="false"',
      sourceElements: '<source />',
      mediaType: 'audio-or-video',
      poster: 'test.jpg',
      title: 'test-title',
      width: 'test-width',
      height: 'test-height',
    }

    const dataFiguresPage = Object.assign({}, data, { inline: false })
    const dataPagebreakBefore = Object.assign({}, data, {
      classes: 'break-before',
    })

    expect(reader.portrait(data)).toMatchSnapshot()
    expect(reader.landscape(data)).toMatchSnapshot()
    expect(reader['portrait-high'](data)).toMatchSnapshot()
    expect(reader.square(data)).toMatchSnapshot()
    expect(reader.audio(data)).toMatchSnapshot()
    expect(reader.video(data)).toMatchSnapshot()
    expect(reader.iframe(data)).toMatchSnapshot()
    expect(reader.vimeo(data)).toMatchSnapshot()
    expect(reader.soundcloud(data)).toMatchSnapshot()

    expect(reader.portrait(dataFiguresPage)).toMatchSnapshot()
    expect(reader.landscape(dataFiguresPage)).toMatchSnapshot()
    expect(reader['portrait-high'](dataFiguresPage)).toMatchSnapshot()
    expect(reader.square(dataFiguresPage)).toMatchSnapshot()
    expect(reader.audio(dataFiguresPage)).toMatchSnapshot()
    expect(reader.video(dataFiguresPage)).toMatchSnapshot()
    expect(reader.iframe(dataFiguresPage)).toMatchSnapshot()

    expect(reader.portrait(dataPagebreakBefore)).toMatchSnapshot()
    expect(reader.landscape(dataPagebreakBefore)).toMatchSnapshot()
    expect(reader['portrait-high'](dataPagebreakBefore)).toMatchSnapshot()
    expect(reader.square(dataPagebreakBefore)).toMatchSnapshot()
    expect(reader.audio(dataPagebreakBefore)).toMatchSnapshot()
    expect(reader.video(dataPagebreakBefore)).toMatchSnapshot()
    expect(reader.iframe(dataPagebreakBefore)).toMatchSnapshot()
  })
})
