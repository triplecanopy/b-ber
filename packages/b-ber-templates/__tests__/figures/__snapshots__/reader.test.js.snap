// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates.reader creates the markup 1`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.reader creates the markup 2`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--landscape test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"landscape\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.reader creates the markup 3`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait-high test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"portrait-high\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.reader creates the markup 4`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"square\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.reader creates the markup 5`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video id=\\"audio-or-video-1\\" autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.reader creates the markup 6`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video id=\\"audio-or-video-1\\" autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.reader creates the markup 7`] = `
"
    
      <div class=\\"figure__large figure__large--iframe\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"iframe\\">
              <iframe
                src=\\"test\\"
                title=\\"test-title\\"
                width=\\"test-width\\"
                height=\\"test-height\\"
                webkitallowfullscreen=\\"webkitallowfullscreen\\"
                mozallowfullscreen=\\"mozallowfullscreen\\"
                allowfullscreen=\\"allowfullscreen\\"
                allow=\\"autoplay\\"
                frameborder=\\"0\\"
                scrolling=\\"no\\"
              />
            </div>

            <div class=\\"figcaption\\">
                <p class=\\"small\\">
                  test<br/>
                  <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.reader creates the markup 8`] = `
"
    
      <div class=\\"figure__large figure__large--vimeo figure__large--undefined\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"vimeo undefined\\">
              <div class=\\"embed unsupported\\">
                <img src=\\"test.jpg\\" alt=\\"Poster Image\\" />
                <iframe
                  data-vimeo=\\"true\\"
                  data-vimeo-poster=\\"test.jpg\\"
                  data-aspect-ratio=\\"undefined\\"
                  src=\\"https://player.vimeo.com/video/testundefined\\" webkitallowfullscreen=\\"webkitallowfullscreen\\" mozallowfullscreen=\\"mozallowfullscreen\\" allowfullscreen=\\"allowfullscreen\\" frameborder=\\"0\\" />
              </div>
            </div>

            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.reader creates the markup 9`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--portrait\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <a href=\\"1.xhtml#ref1\\">
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test\\"/>
    </a>
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.reader creates the markup 10`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--landscape\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <a href=\\"1.xhtml#ref1\\">
      <img class=\\"landscape\\" alt=\\"test\\" src=\\"../images/test\\"/>
    </a>
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.reader creates the markup 11`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--portrait-high\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <a href=\\"1.xhtml#ref1\\">
      <img class=\\"portrait-high\\" alt=\\"test\\" src=\\"../images/test\\"/>
    </a>
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.reader creates the markup 12`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--square\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <a href=\\"1.xhtml#ref1\\">
      <img class=\\"square\\" alt=\\"test\\" src=\\"../images/test\\"/>
    </a>
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.reader creates the markup 13`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video id=\\"audio-or-video-1\\" autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    </section>
    "
`;

exports[`templates.reader creates the markup 14`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video id=\\"audio-or-video-1\\" autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    </section>
    "
`;

exports[`templates.reader creates the markup 15`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--iframe\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"iframe\\">
              <iframe
                src=\\"test\\"
                title=\\"test-title\\"
                width=\\"test-width\\"
                height=\\"test-height\\"
                webkitallowfullscreen=\\"webkitallowfullscreen\\"
                mozallowfullscreen=\\"mozallowfullscreen\\"
                allowfullscreen=\\"allowfullscreen\\"
                allow=\\"autoplay\\"
                frameborder=\\"0\\"
                scrolling=\\"no\\"
              />
            </div>

            <div class=\\"figcaption\\">
                <p class=\\"small\\">
                  test<br/>
                  <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    </section>
    "
`;

exports[`templates.reader creates the markup 16`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.reader creates the markup 17`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--landscape break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"landscape\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.reader creates the markup 18`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait-high break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"portrait-high\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.reader creates the markup 19`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"square\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.reader creates the markup 20`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video id=\\"audio-or-video-1\\" autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.reader creates the markup 21`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video id=\\"audio-or-video-1\\" autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.reader creates the markup 22`] = `
"
    
      <div class=\\"figure__large figure__large--iframe\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"iframe\\">
              <iframe
                src=\\"test\\"
                title=\\"test-title\\"
                width=\\"test-width\\"
                height=\\"test-height\\"
                webkitallowfullscreen=\\"webkitallowfullscreen\\"
                mozallowfullscreen=\\"mozallowfullscreen\\"
                allowfullscreen=\\"allowfullscreen\\"
                allow=\\"autoplay\\"
                frameborder=\\"0\\"
                scrolling=\\"no\\"
              />
            </div>

            <div class=\\"figcaption\\">
                <p class=\\"small\\">
                  test<br/>
                  <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;
